AirSkyBoat/AirSkyBoat
# Bug fix/Make OAX weapons work with Jump

**_I affirm:_**


## Please enter a player-facing description


* Fixed WS calculation for Jump type abilities to take "Occasionally Attacks X" weapons into account. (Both MainHand and OffHand) (HeavensSword)
* Fixed Jump TP calculation when multi-hits occur. (HeavensSword)
* Made it so Jump abilities will no longer give TP to the target. (This is how retail functions) (HeavensSword)

## What does this pull request do? (Please be technical)


* WS can now utilize a new parameter "canUseOccAtkX". Currently this is only needed for Jump abilities, but I went with a more flexible approach in case a use case arises where a WS or Ability that utilizes WS DMG calcs like Jump need to respect OAX Weapons.
* I've exposed the GetHitCount method of CItemWeapon through the BattleEntity so that it can now be called from Lua. 
* Previously the Jump multi-hit TP calculation was incorrectly changing the tpMultiplier which could result in higher than expected values. I've changed it to instead collate the total hits into tpHitsLanded.
*  The last major change was adding a new argument to the takeWeaponskillDamage methods that is just a boolean for whether the WS was a Jump ability. This is necessary because Jumps do not actually give TP to defenders on retail, so a check has been added to avoid that happening.
* Translated a number of the old Russian code comments for clarity.

## Steps to test these changes


The easiest way to test these changes is to make your GM character a WAR/DRG and equip a Ridill. (Avoid using Store TP Gear)
You should then be able to use Jump and High Jump on a mob and observe your TP gain. I believe for Ridill's delay, the base gain from 1 hit is 63, but you can figure it out by either waiting for a single auto-attack, or switch to a Hofud and observe the TP gain to get your base.
You should see 63 TP * the number of OAX hits for that activation. 
EDIT: If DA/TA/QA proc, they take precedence over OAX calculation.

If you want some fun, switch to NIN/DRG and equip Kraken Club and Merc. Kris. This helps validate the behavior with Dual Wield.

## Special Deployment Considerations




There shouldn't be any special considerations.
